### DISCUSSION TITLE
Auto Upgrade with Pre Downloaded Media (ISO or Window Files)

### DISCUSSION THREAD

**Discussion by TTonsfeldt** [AveYo/MediaCreationTool.bat#129] [General]
The MediaCreationTool has been great however when running this on multiple machines logged on as the same user it causes some issues with the naming of the ISO when auto downloading the iso along with an long time to download on each device the same file

Is their an way to have an pre downloaded iso used for the Auto Upgrade part of the script instead of downloading each time or can the script save the files differently that doesn't cause an conflict when using the same account?

**Comment by AveYo** [comment]
`auto.cmd` gets included on the created media specifically for this purpose!
Just create USB and plug it into another PC or create ISO and mount it (with double-click) on other PC 
Then run `auto.cmd` instead of `setup.exe`

**Reply by TTonsfeldt** [reply]
Can the auto.cmd be ran from an ISO on an network location or on the C Drive of the device besides from an USB?

Looking at upgrading over 400 devices and going to each one and plugging USB could be done but takes time,  where i can push out an package using PDQ Deploy to upgrade but it had issues since last using it for 1909 upgrade in 2019

**Reply by AveYo** [reply]
sure, it runs just fine from a network share!

**Reply by AveYo** [reply]
note that `auto.cmd` has been enhanced some more!
from the new script you can just copy-paste the generate_auto_cmd section into a new file,  
or you can start auto usb preset then close setup window, and auto.cmd gets generated in `C:\ESD\MCT`
